src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -11489,6 +11489,103 @@ class Enumerator1
                 );
         }
 
+        [Fact]
+        [WorkItem(64218, "https://github.com/dotnet/roslyn/issues/64218")]
+        public void LocalScope_12_Foreach_03()
+        {
+            var source =
+@"class Program
+{
+    static void Main()
+    {
+        int i0 = 0;
+        S s0 = new S(ref i0);
+        {
+            int i1 = 1;
+            foreach (S s1 in Enumerable1.Create(ref i1)) {
+                s0 = s1;
+                break;
+            }
+        }
+    }
+}
+ref struct S
+{
+    public S(ref int i) { }
+}
+ref struct Enumerable1
+{
+    public static Enumerable1 Create(ref int p) => default;
```

Comment:
>public static Enumerable1 Create(ref int p) => default; [](http://example.com/codeflow?start=4&length=55)

There is a number of other "foreach" tests that use either `ref` or ref-struct parameter  for ```Create``` method. Should we add flavors of those where Enumerable is a ref-struct?

Reply:
OK, I added ref-struct flavors that I think make most sense (in particular, such tests that have an assignment inside the `foreach`): 7898375b7880b72a6ea8c477b444d41558d98599